src/ui/common/src/components/integrations/integrationObjectPreview.tsx

```diff
@@ -0,0 +1,42 @@
+import { Alert, Typography } from '@mui/material';
+import Box from '@mui/material/Box';
+import CircularProgress from '@mui/material/CircularProgress';
+import React from 'react';
+
+import DataTable from '../../components/tables/dataTable';
+import { ObjectState } from '../../reducers/integration';
+import { isFailed, isLoading, isSucceeded } from '../../utils/shared';
+
+type Props = {
+  objectName: string;
+  object: ObjectState;
+};
+
+const IntegrationObjectPreview: React.FC<Props> = ({ objectName, object }) => {
+  return (
+    <Box sx={{ mt: 3 }}>
+      {isLoading(object.status) && (
+        <Box sx={{ display: 'flex', flexDirection: 'row', mt: 3 }}>
+          <CircularProgress size={30} />
+          <Typography sx={{ ml: 2 }}>
+            Loading table <b>{objectName}</b>...
```

Comment:
Wonder if we should update the terminology so it says "Loading object" rather than table. Table is currently the only things we are previewing so not sure if that is necessary.

Reply:
nice catch, updated